src/main/java/core/basesyntax/operations/BalanceOperation.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+    private final Storage storage;
+
+    public BalanceOperation(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (storage == null) {
+            throw new IllegalArgumentException("Storage cannot be null.");
+        }

```

Comment:
You don't need this check because you are passing this variable as the developer, not the user. You don't need to check yourself, do you? ))